### ISSUE TITLE
Import resolution issue with `extraPaths` when main path has `__init__.py`

### ISSUE THREAD

**Issue by 0xjc** [microsoft/pyright#5203]
**Describe the bug**
When a directory contains `__init__.py`, sibling files in a corresponding `extraPaths` path fail to resolve when imported.

**To Reproduce**
Create the following directory structure:
```
./
├── extra/
│   └── foo/
│       └── bar.py
├── foo/
│   └── __init__.py
├── main.py
└── pyrightconfig.json
```

`extra/foo/bar.py`:
```
class Bar:
    pass
```

`foo/__init__.py`: empty

`main.py`:
```
from foo.bar import Bar
```

`pyrightconfig.json`:
```
{
  "executionEnvironments": [
    {
      "root": ".",
      "extraPaths": [
        "extra"
      ]
    }
  ]
}
```

Then run: `pyright main.py`

**Expected behavior**
Expected no errors, but got:
```
/tmp/example/main.py
  /tmp/example/main.py:1:6 - error: Import "foo.bar" could not be resolved (reportMissingImports)
1 error, 0 warnings, 0 informations
```

Note that if the empty file `foo/__init__.py` is deleted, then Pyright passes with no errors.

**Screenshots or Code**
See above

**VS Code extension or command-line**
Command-line, pyright v1.1.310

**Additional context**
None

**Comment by erictraut** [comment]
This current behavior is intended.

Pyright models the behavior of the Python import loader which prefers traditional packages (those with `__init__.py` files) over implicit namespace packages (those without `__init__.py` files). If we were to deviate from this runtime behavior, it would create false positives (and false negatives) in various import resolution scenarios.

You'll need to find a workaround in your code if you want this to work. For example, you could add an empty `__init__.py` file under the `extra/foo` directory.
